### PR TITLE
Iron

### PR DESCRIPTION
**Pull request by dbrgn**
Moved from Hyper to [Iron](http://ironframework.io/doc/iron/index.html).

It's more high level and will allow us to easily do routing.

@rnestler can you review?

### CODE REVIEW

File: spaceapi_server/src/lib.rs [hunk 1]
```diff
@@ -140,23 +138,16 @@ fn status_endpoint(_: Request, mut res: Response<Fresh>) {
...
+    // Create response
+    let mut response = Response::with((status::Ok, body));
 
-    // Write response body
-    let mut stream = res.start().unwrap();
-    stream.write_all(body_bytes).unwrap();
+    // Set headers
+    response.set_mut(Header(headers::ContentType("application/json; charset=utf-8".parse().unwrap())));
+    response.set_mut(Header(headers::CacheControl(vec![headers::CacheDirective::NoCache])));
+    response.set_mut(Header(headers::AccessControlAllowOrigin::Any));
```
Comment (dbrgn) [inline spaceapi_server/src/lib.rs]: Here, instead of using `set_mut` we could also use chained `set` calls on the response, in which case it wouldn't need to be mutable.

But this way it's a bit nicer to structure the code.

If you think the chained approach would be better, let me know.

Comment (rnestler) [inline spaceapi_server/src/lib.rs]: I think whats more concerning are the various ``unwrap()`` calls. That way the server will just panic if anything goes wrong. I wouldn't care if it panics at startup, but while handling requests...
Maybe we could cache things like ``headers::ContentType("application/json; charset=utf-8".parse().unwrap()`` at startup so we would fail immediately not after handling the first request.

Comment (dbrgn) [inline spaceapi_server/src/lib.rs]: Yes, that's a good point.

Once more, it would be great if we can have a Server struct where we can initialize things like these headers from the constructors. The move to Iron should simplify that.

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ services:
 env:
   - CRATE=.
   - CRATE=spaceapi
+  - CRATE=spaceapi_server
 script: cd $CRATE && cargo build --verbose && cargo test --verbose
```
Comment (rnestler) [inline .travis.yml]: This is not Iron specific is it? You could commit it separately and directly into the rust branch.

Comment (dbrgn) [inline .travis.yml]: It fixes the tests for the iron branch. Previously, tests were not run.

Yes, I could have put this in a separate pull request, as it's independent. But that would have been a bit of overkill :) As long as it's a separate commit I think it's fine.

Comment (rnestler) [inline .travis.yml]: No need to rebase, I just needed something to nag about ;)

**Comment by rnestler** [comment]
I think code wise It looks good and seems to simplify things. What I don't like is that cb93044188c11c2ed5ab02a7ce8a23b72472bfba comments some code and 05c653f0c1c24e95be648fccc198652e7fbccb5e deletes and replaces the commented code. But thats just me nagging again ;)

**Comment by dbrgn** [comment]
> What I don't like is that cb93044 comments some code and 05c653f deletes and replaces the commented code.

Well, but THAT'S to keep commits small and focussed. First replace the server, then the header configuration ;)
